business/checkers/destinationrules/traffic_policy_checker.go

```diff
@@ -55,6 +58,31 @@ func (t TrafficPolicyChecker) drsWithNonLocalmTLSEnabled() []kubernetes.IstioObj
 	return mtlsDrs
 }
 
+func sameHostDestinationRules(dr kubernetes.IstioObject, mdrs []kubernetes.IstioObject) []kubernetes.IstioObject {
+	var drHost kubernetes.Host
+	shdrs := make([]kubernetes.IstioObject, 0, len(mdrs))
+
+	if host, ok := dr.GetSpec()["host"]; ok {
+		if dHost, ok := host.(string); ok {
+			drHost = kubernetes.ParseHost(dHost, dr.GetObjectMeta().Namespace, dr.GetObjectMeta().ClusterName)
```

Comment:
Maybe this was just some environments (or tests only?), but the ClusterName was in some cases empty..

Reply:
Arr, nm.. the ParseHost takes care of that.

Reply:
yes, I assume that ParseHost it will populate those 3 fields(service, ns, cluster). Except for "*.local" that it only will have service with "*.local" value.